edocumentation-service/src/main/java/org/dcsa/edocumentation/service/unofficial/BookingValidationService.java

```diff
@@ -71,11 +71,75 @@ private String validateBooking(Booking booking) {
     if (isInThePast(today, booking.getExpectedArrivalAtPlaceOfDeliveryEndDate())) {
       return "expectedArrivalAtPlaceOfDeliveryEndDate is in the past";
     }
-
-    return null;
+    return validateShipmentLocations(booking.getShipmentLocations());
   }
 
   private boolean isInThePast(LocalDate today, LocalDate time) {
     return time != null && today.isAfter(time);
   }
+
+  private String validateShipmentLocations(Set<ShipmentLocation> shipmentLocations) {
+    if (shipmentLocations.isEmpty()) {
+      return "Invalid booking: Shipment locations should not be empty";
+    }
+
+    boolean hasPREorPOL =
+        shipmentLocations.stream()
+            .map(ShipmentLocation::getShipmentLocationTypeCode)
+            .anyMatch(code -> code == LocationType.PRE || code == LocationType.POL);
+
+    if (!hasPREorPOL) {
+      return "No ShipmentLocationTypeCode of PRE or POL found in the shipmentLocations, "
+          + "either one should be provided both is allowed";
+    }
+
+    boolean hasPODorPDE =
+        shipmentLocations.stream()
+            .map(ShipmentLocation::getShipmentLocationTypeCode)
+            .anyMatch(code -> code == LocationType.POD || code == LocationType.PDE);
+
+    if (!hasPODorPDE) {
+      return "No ShipmentLocationTypeCode of POD or PDE found in the shipmentLocations, "
+          + "either one should be provided both is allowed";
+    }
+
+    var filteredByUNLocationCode =
+        shipmentLocations.stream()
+            .filter(sl -> sl.getLocation().getUNLocationCode() != null)
+            .toList();
```

Comment:
I think we were unclear in the issue and in my clarification. The distinct logic was only defined for the `PRE`, `POL`, `POD` and `PDE` locations.

(We have not determined if other shipment locations can cause duplicate UN Location code)